pyproject.toml

```diff
@@ -20,16 +20,16 @@ Jinja2 = "==3.1.1"
 MarkupSafe = "==2.1.1"
 packaging = "==21.3"
 prompt-toolkit = "==3.0.28"
-python = "^3.7"
-PyYAML = "==6.0"
+python = "==3.9.10"
```

Comment:
With the previous version, `pyinstaller` was failing to resolve when running `poetry lock` due to the more permissive `python` version declared here. 

Additionally, it seems to make sense to pin to `3.9.10` since this is what `hokusai` is [tested on and the runtime version we explicitly declare](https://github.com/search?q=repo%3Aartsy%2Fhokusai%203.9.10&type=code).

Reply:
This is causing problems [here](https://app.circleci.com/pipelines/github/artsy/horizon/2884/workflows/10e9a820-6eb7-4589-b4ea-a58af3878099/jobs/3400) because the _precise_ python version isn't available in alpine. What if we switched to something like `^3.9`?

Reply:
let's do it (or even `^3.8` if we want to be safe). and watch for whether the `pyinstaller` problem recurs.

Reply:
https://github.com/artsy/hokusai/pull/366